Bio/SeqIO/Interfaces.py

```diff
@@ -128,30 +136,32 @@ class SequenceWriter:
     the number of records.
     """
 
-    def __init__(self, target, mode="w"):
+    def __init__(self, target: Union[str, IO[Any]], mode: str = "w") -> None:
         """Create the writer object."""
         if mode == "w":
-            try:
-                target.write("")
-            except TypeError:
-                # target was opened in binary mode
-                raise StreamModeError("File must be opened in text mode.") from None
-            except AttributeError:
+            if isinstance(target, str):
```

Comment:
Was switching this round to do type checking deliberate? If so, why?

Reply:
Yes, this fixed typing errors and disambiguated `target` which can be a string or an open file.

Reply:
Thank you for clarifying. So other bits of code which have arguments accepting multiple distinct types might need changes from duck-typing (does it have a ``.write(...)`` method) to explicit type checking.

Reply:
Duck typing can still work with mypy but it can get messy. Exceptions are also expensive for a common code path.